1.19.3 missing sound packet field

```
[02:00:08 WARN]: [OldCombatMechanics] Error detecting sound packets. Please report it along with the following exception on github.
java.lang.ExceptionInInitializerError: null
        at kernitus.plugin.OldCombatMechanics.utilities.packet.sound.SoundPacket.getSoundName(SoundPacket.java:27) ~[OldCombatMechanics.jar:?]
        at kernitus.plugin.OldCombatMechanics.module.ModuleAttackSounds$SoundListener.lambda$onPacketSend$0(ModuleAttackSounds.java:79) ~[OldCombatMechanics.jar:?]
        at java.util.Optional.filter(Optional.java:218) ~[?:?]
        at kernitus.plugin.OldCombatMechanics.module.ModuleAttackSounds$SoundListener.onPacketSend(ModuleAttackSounds.java:79) ~[OldCombatMechanics.jar:?]
        at kernitus.plugin.OldCombatMechanics.utilities.packet.mitm.PacketInjector.write(PacketInjector.java:187) ~[OldCombatMechanics.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:764) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:790) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:758) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1020) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:311) ~[netty-transport-4.1.82.Final.jar:4.1.82.Final]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelProxy.writeAndFlush(NettyChannelProxy.java:227) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelProxy.writeAndFlush(NettyChannelProxy.java:233) ~[ProtocolLib.jar:?]
        at net.minecraft.network.Connection.doSendPacket(Connection.java:459) ~[?:?]
        at net.minecraft.network.Connection.lambda$sendPacket$11(Connection.java:433) ~[?:?]
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.lambda$proxyRunnable$2(NettyEventLoopProxy.java:48) ~[ProtocolLib.jar:?]
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403) ~[netty-transport-classes-epoll-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar:4.1.82.Final]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.RuntimeException: Field with type SoundEffect not found
        at kernitus.plugin.OldCombatMechanics.utilities.reflection.Reflector.getFieldByType(Reflector.java:159) ~[OldCombatMechanics.jar:?]
        at kernitus.plugin.OldCombatMechanics.utilities.packet.sound.SoundPacket$PacketAccess.<clinit>(SoundPacket.java:55) ~[OldCombatMechanics.jar:?]
        ... 23 more
```

--- a/src/main/java/kernitus/plugin/OldCombatMechanics/utilities/packet/sound/SoundPacket.java
+++ b/src/main/java/kernitus/plugin/OldCombatMechanics/utilities/packet/sound/SoundPacket.java
@@ -12,23 +12,25 @@ import kernitus.plugin.OldCombatMechanics.utilities.reflection.Reflector;
 import kernitus.plugin.OldCombatMechanics.utilities.reflection.type.ClassType;
 
 import java.lang.reflect.Field;
+import java.lang.reflect.Method;
 import java.util.Optional;
 
 public class SoundPacket implements ImmutablePacket {
     private static final Class<?> PACKET_CLASS = PacketHelper.getPacketClass(PacketType.PlayOut, "NamedSoundEffect");
+    private static final PacketAccess PACKET_ACCESS = PacketAccess.getAccess();
 
     private final Object nmsPacket;
 
-    private SoundPacket(ImmutablePacket packet){
+    private SoundPacket(ImmutablePacket packet) {
         this.nmsPacket = packet.getNmsPacket();
     }
 
-    public String getSoundName(){
-        return PacketAccess.getSoundName(getNmsPacket());
+    public String getSoundName() {
+        return PACKET_ACCESS.getSoundName(getNmsPacket());
     }
 
     @Override
-    public Object getNmsPacket(){
+    public Object getNmsPacket() {
         return nmsPacket;
     }
 
@@ -38,27 +40,67 @@ public class SoundPacket implements ImmutablePacket {
      * @param packet the packet to convert
      * @return the sound packet or empty if it had thr wrong type
      */
-    public static Optional<SoundPacket> from(ImmutablePacket packet){
-        if(packet.getPacketClass() == PACKET_CLASS){
+    public static Optional<SoundPacket> from(ImmutablePacket packet) {
+        if (packet.getPacketClass() == PACKET_CLASS) {
             return Optional.of(new SoundPacket(packet));
         }
         return Optional.empty();
     }
 
-    private static class PacketAccess {
+    private interface PacketAccess {
+        String getSoundName(Object packet);
+
+        static PacketAccess getAccess() {
+            try {
+                return new PacketAccessPreV1_19_3();
+            } catch (Throwable igored) {
+                try {
+                    return new PacketAccessV1_19_3();
+                } catch (Throwable e) {
+                    throw new RuntimeException("Sound packet init failed", e);
+                }
+            }
+        }
+    }
+
+    private static class PacketAccessV1_19_3 implements PacketAccess {
         private static final Class<?> SOUND_EFFECT_CLASS = Reflector.getClass(ClassType.NMS, "sounds.SoundEffect");
 
-        private static final Field SOUND_EFFECT_FIELD;
-        private static final Field MINECRAFT_KEY_FIELD;
+        private final Field soundEffectHolderField;
+        private final Method soundEffectMethod;
+        private final Field minecraftKeyField;
 
-        static{
-            SOUND_EFFECT_FIELD = Reflector.getFieldByType(PACKET_CLASS, "SoundEffect");
-            MINECRAFT_KEY_FIELD = Reflector.getFieldByType(SOUND_EFFECT_CLASS, "MinecraftKey");
+        private PacketAccessV1_19_3() {
+            soundEffectHolderField = Reflector.getFieldByType(PACKET_CLASS, "Holder");
+
+            Class<?> holderClass = Reflector.getClass(ClassType.NMS, "core.Holder");
+            soundEffectMethod = Reflector.getMethodByGenericReturnType(holderClass.getTypeParameters()[0], holderClass);
+            minecraftKeyField = Reflector.getFieldByType(SOUND_EFFECT_CLASS, "MinecraftKey");
         }
 
-        public static String getSoundName(Object packet){
-            Object soundEffect = Reflector.getUnchecked(() -> SOUND_EFFECT_FIELD.get(packet));
-            Object minecraftLey = Reflector.getUnchecked(() -> MINECRAFT_KEY_FIELD.get(soundEffect));
+        public String getSoundName(Object packet) {
+            Object holder = Reflector.getUnchecked(() -> soundEffectHolderField.get(packet));
+            Object soundEffect = Reflector.getUnchecked(() -> soundEffectMethod.invoke(holder));
+            Object minecraftLey = Reflector.getUnchecked(() -> minecraftKeyField.get(soundEffect));
+
+            return minecraftLey.toString();
+        }
+    }
+
+    private static class PacketAccessPreV1_19_3 implements PacketAccess {
+        private static final Class<?> SOUND_EFFECT_CLASS = Reflector.getClass(ClassType.NMS, "sounds.SoundEffect");
+
+        private final Field soundEffectField;
+        private final Field minecraftKeyField;
+
+        private PacketAccessPreV1_19_3() {
+            soundEffectField = Reflector.getFieldByType(PACKET_CLASS, "SoundEffect");
+            minecraftKeyField = Reflector.getFieldByType(SOUND_EFFECT_CLASS, "MinecraftKey");
+        }
+
+        public String getSoundName(Object packet) {
+            Object soundEffect = Reflector.getUnchecked(() -> soundEffectField.get(packet));
+            Object minecraftLey = Reflector.getUnchecked(() -> minecraftKeyField.get(soundEffect));
 
             return minecraftLey.toString();
         }

--- a/src/main/java/kernitus/plugin/OldCombatMechanics/utilities/reflection/Reflector.java
+++ b/src/main/java/kernitus/plugin/OldCombatMechanics/utilities/reflection/Reflector.java
@@ -117,6 +117,16 @@ public class Reflector {
                 .orElse(null);
     }
 
+    public static Method getMethodByGenericReturnType(TypeVariable<?> typeVar, Class<?> clazz){
+        for (Method method : clazz.getMethods()){
+            if (method.getGenericReturnType().getTypeName().equals(typeVar.getName())){
+                return method;
+            }
+        }
+        throw new RuntimeException("Method with type " + typeVar + " not found");
+    }
+
+
     public static <T> T invokeMethod(Method method, Object handle, Object... params) {
         try {
             @SuppressWarnings("unchecked")

